qep-314-coding-style.md

```diff
@@ -78,7 +78,21 @@ Notes:
 - 3.1. For readability and ease of code review, avoid use of ``auto``. The following exceptions are permitted:
 
   - 3.1.1. ``auto`` should be used for complex types, such as iterators. Eg ``for ( auto it = object.begin(); ...)``
-  
+  - 3.1.2. ``auto`` may be used for variable types if the type is explicit during variable initialization. Eg
+
+```
+// allowed, as the QgsPoint type is explicit during initialization:
+auto pointObject = QgsPoint( 3, 4 );
```

Comment:
Is there any reason we would want to write this instead of `QgsPoint pointObject( 3, 4 );` ? The notation with `auto` is longer, and may involve extra copy of the temporary object?

Reply:
@wonder-sk 

> Is there any reason we would want to write this instead of QgsPoint pointObject( 3, 4 ); ? The notation with auto is longer, and may involve extra copy of the temporary object?

That's a good point. Should we reword this change to only allow for some explicit types like std::unique_ptr / std::shared_ptr ?

Reply:
> Should we reword this change to only allow for some explicit types like std::unique_ptr / std::shared_ptr ?

yeah I think that would be good...